src/types.ts

```diff
@@ -0,0 +1,50 @@
+import NetError from './error';
+
+export type Methods = 'DELETE' | 'GET' | 'HEAD' | 'OPTIONS' | 'PATCH' | 'POST' | 'PUT';
+
+export type QueryParams = string | [string, any][] | Record<string, any>;
+
+export type RequestHeaders = [string, any][] | Record<string, any>;
+
+export type Events = {
+  complete?<T>(response: Response<T>): void;
+  error?(error: Error): void;
+  downloadProgress?(event: ProgressEvent): void;
+  uploadProgress?(event: ProgressEvent): void;
+};
+
+export type Transformers<T> = {
+  request?(body: T | any): any;
+  response?(data: any): T;
```

Comment:
This response transformer must return the same type as the request body?

Reply:
Oh, that's wrong, good catch.

Reply:
Oh boy, so fixing this created a much larger change on the api repo side!  But typing / usage of this is much more solid now.